easygcm-sample/src/main/java/eu/inloop/easygcm/sample/App.java

```diff
@@ -4,22 +4,20 @@
 import android.os.Bundle;
 
 import eu.inloop.easygcm.GcmListener;
-import eu.inloop.easygcm.WakeLockRelease;
 
 public class App extends Application implements GcmListener {
 
     @Override
-    public void onMessage(String s, Bundle bundle, WakeLockRelease wakeLockRelease) {
-        System.out.println("### message: " + s);
+    public void onMessage(String from, Bundle data) {
```

Comment:
As we found in documentation, the public void onMessageReceived(String from, Bundle data) method is run in an asynctask and it's ensured that Android will not kill it while AsyncTask is running. The library should state that the onMessage callback will be called on non-ui thread and should not launch new AsyncTasks, it should call networks etc. directly.